docs/testing-styleguide.MD

```diff
@@ -0,0 +1,280 @@
+      | large  | 48     |
+
+  @positive 
+  Scenario: Verify the click function for a Button component
+    Given clear all actions in Actions Tab
+    When I click on "button"
+    Then click action was called in Actions Tab
+```
+
+##### Locators
+Moving element locators into locator files make the step definition files easier to read. Step definition files call `index.js` files. `index.js` files contain exported arrow functions for locators. They import information from `locator.js` files. `locator.js` files contain constant string selectors for locating elements.
+
+Each component has their own `index.js` and `locators.js` files stored in `carbon/cypress/locators/component-name` unless the locators are common to more than one component/element/story, in which case they are contained in `carbon/cypress/locators`.
+
+###### How to locate elements:
+Always use unique selectors to locate elements with the order below:
+1.	`[data-component="unique-tag"]` for component and `[data-element="unique-tag"]` for component's element
+2.	`#id`
+3.	`.class-name`
+
+###### How to name locators:
+*  Component on preview - `componentPreview`
+*  Knobs - `componentKnobs`
+
+An example of an `index.js` file:
+```
+import { BUTTON_SUBTEXT_PREVIEW, BUTTON_DATA_COMPONENT_PREVIEW } from './locators';
+
+// component preview locators
+export const buttonSubtextPreview = () => cy.iFrame(BUTTON_SUBTEXT_PREVIEW);
+export const buttonDataComponent = () => cy.iFrame(BUTTON_DATA_COMPONENT_PREVIEW);
+
+// component preview locators in no IFrame
+export const buttonDataComponentNoIFrame = () => cy.get(BUTTON_DATA_COMPONENT_PREVIEW);
+```
+
+An example of a corresponding `locators.js` file:
+
+```
+// component preview locators
+export const BUTTON_SUBTEXT_PREVIEW = '[data-element="subtext"]';
+export const BUTTON_DATA_COMPONENT_PREVIEW = '[data-component="button"]';
+```
+
+Other support files include the `helper.js` file which contains functions such as that used to prepare the URL of the component page and the control of draggable components.
+
+
+##### Test Results
+Depending on how Cypress was run, you will find test results in either the Cypress UI, the console or in GitHub Actions linked from the build checks in GitHub. For further information see the [Getting Started Guide](https://github.com/Sage/carbon/blob/master/cypress/README.md).
+
+##### Applitools
+###### Integration with Cypress
+Instructions for adding Applitools to an existing Cypress project can be found [here](https://applitools.com/tutorials/cypress.html). Applitools is integrated with Cypress by means of various configuration and settings in the `cypress/support/index.js` file, `applitools.config.js` file, and with additional feature files and corresponding step definitions of those components/stories where we want to do visual testing. When the relevant tests are run from Cypress, snapshots will be taken where specified in the feature files. We currently run visual tests in Chrome browser only.
```

Comment:
I think we need also to mention about using only `chrome` as a browser in `applitools` checks.